.github/workflows/gpuci.yml

```
@@ -0,0 +1,61 @@
+name: "GPU CI/CD"
+
+on:
+  pull_request:
+    branches:
+      - 'main'
+      - '[rv][0-9].[0-9].[0-9]'
+      - '[rv][0-9].[0-9].[0-9]rc[0-9]'
+    types: [ labeled ]
+
+jobs:
+  build-container:
+    if: ${{ github.event.label.name == 'gpuci' }}
+    uses: ./.github/workflows/_build_container.yml
+
+  gpu-test:
+    needs: build-container
+    runs-on: self-hosted
+    if: ${{ github.event.label.name == 'gpuci' }}
+
+    steps:
+    - name: Run nvidia-smi test
+      run: |
+        whoami
+        nvidia-smi
+
+    - name: Checkout the PR code
+      uses: actions/checkout@v4
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Pull the Docker image built in the previous step
+      run: docker pull nemoci.azurecr.io/nemo_curator_container:buildcache
+
+    - name: Run Docker container with Python 3.10
+      run: |
+        docker run --name nemo-curator-container -d nemoci.azurecr.io/nemo_curator_container:buildcache bash -c "sleep infinity"
```

```suggestion
        docker run --name nemo-curator-container -d nemoci.azurecr.io/nemo_curator_container:${{ github.run_id }} bash -c "sleep infinity"
```
Very important note: The image tag `buildcache` is only meant to be used for speeding up builds as it provides a cache manifest. The source code might be outdated (which is okay for a cache image since it will lead to a cache miss at the relevant layer). But for running your test suite you want to use the produced image which you can find by the image tag `${{ github.run_id }}`